contracts/dual_token/DualStakingToken.sol

```diff
@@ -104,14 +104,31 @@ contract DualStakingToken is ERC721TokenReceiver {
         secondTokenAmount = (minStakingDeposit * (1 + numAgentInstances) * _rewardRatio) / 1e18;
     }
 
+    /// @dev Claims second token reward.
+    /// @param multisig Service multisig address.
+    /// @param reward Second token non-zero reward.
+    function _claim(address multisig, uint256 reward) internal {
+        // Recalculate reward in second token value
+        reward = (reward * rewardRatio) / 1e18;
+
+        // Transfer second token reward to the service multisig
+        // Get second token balance, reserving the staked amount untouched
+        uint256 balance = IToken(secondToken).balanceOf(address(this)) - secondTokenAmount * numServices;
+
+        // Limit reward if there is not enough on a balance
+        if (reward > balance) {
+            reward = balance;
+        }
+
+        // Withdraw reward to service multisig
```

Comment:
need a zero check here on "reward" to avoid 0 transfers

Reply:
This function is only called when reward is not zero. I'll add a note.